src/cpu/x64/matmul_inner_product.cpp

```diff
@@ -0,0 +1,271 @@
+/*******************************************************************************
+* Copyright 2025 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <map>
+
+#include "cpu/x64/matmul_inner_product.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace x64 {
+
+status_t create_matmul_pd(std::shared_ptr<primitive_desc_t> &matmul_pd,
+        engine_t *engine, const memory_desc_t *a_md, const memory_desc_t *b_md,
+        const memory_desc_t *c_md, const memory_desc_t *ip_bia_md,
+        const primitive_attr_t *attr) {
+    auto matmul_desc = matmul_desc_t();
+
+    const bool with_bias = bool(ip_bia_md);
+
+    memory_desc_t mm_bia_md {};
+    // Inner Product bias is always a vector while MatMul requires bias to have
+    // the same number of dimensions as that of the output tensor, therefore an
+    // adjustment is required.
+    if (with_bias) {
+        assert(ip_bia_md->ndims == 1);
+        dims_t mm_bia_dims = {1, ip_bia_md->dims[0]};
+        CHECK(memory_desc_init_by_tag(mm_bia_md, 2, mm_bia_dims,
+                ip_bia_md->data_type, format_tag::ab));
+    }
+
+    CHECK(matmul_desc_init(
+            &matmul_desc, a_md, b_md, with_bias ? &mm_bia_md : nullptr, c_md));
+
+    primitive_desc_iterator_t it(
+            engine, (op_desc_t *)&matmul_desc, attr, nullptr);
+
+    matmul_pd = *(++it);
+    if (!matmul_pd) return status::unimplemented;
+
+    return status::success;
+}
+
+status_t init_matmul_md(memory_desc_t &mm_md, const memory_desc_t &ip_md,
+        format_tag_t tag, bool swap_dims) {
+    auto p_dims = ip_md.dims;
+    auto p_dim1 = utils::array_product(p_dims + 1, ip_md.ndims - 1);
+
+    if (swap_dims) {
+        dims_t dims_2d = {p_dim1, p_dims[0]};
```

Comment:
?
```suggestion
        dims_t dims_2d = (swap_dims ? {p_dim1, p_dims[0]} : {p_dims[0], p_dim1});
```

Reply:
This syntax is not legal in the language. The closest thing we could get is using the ternary operator for each dimension but the code would be hard to read.
`dims_t dims_2d = {swap_dims ? p_dim1 : p_dims[0], swap_dims ? p_dims[0] : p_dim1};`